src/ledger/backend/readconn.ts

```diff
@@ -12,15 +15,20 @@ export const ErrErdstallContractNotConnected = new Error(
 	"erdstall contract not connected",
 );
 
-export interface LedgerReader extends ErdstallWatcher, Contracter {}
+export interface LedgerReader
+	extends NFTMetadataProvider,
+		ErdstallWatcher,
+		Contracter {}
 
 export class LedgerReadConn implements LedgerReader {
 	readonly contract: Erdstall;
 	private eventCache: Map<Function, (args: Array<any>) => void>;
+	private metadataUriCache: Map<string, NFTMetadata>;
```

Comment:
It really is a cache for metadata itself, not for the URIs. So either rename it to `metadataCache` or if your intention was to only cache URIs, then you need to change the logic in `getNftMetadata` and change the Map value type to strings/URIs.